core/arch/arm/kernel/entry_a64.S

```diff
@@ -29,8 +30,7 @@
 		adr_l	x1, stack_tmp_stride
 		ldr	w1, [x1]
 		mul	x1, x0, x1
-		adr_l	x0, stack_tmp_export
-		ldr	x0, [x0]
+		bl	thread_get_stack_tmp_export
```

Comment:
SP is unassigned at this point. How can you know that `thread_get_stack_tmp_export()` doesn't use the stack?

Reply:
Good point. It happens to be the case (I checked the disassembly), but there is obviously no guarantee so I need to find another way. I think I'll rewrite the function in assembly unless you have a better idea.

Reply:
In that case we could just add the needed instructions here.